### PR TITLE
feat: add sdk docs generation workflow

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
We will follow similar patterns to the sdk action but this action will be a little simpler
- Support PR Mode
- Support Direct Mode
- Validate the spec before docs generation
- Compile docs before finalizing

### CODE REVIEW

File: .github/workflows/sdk-docs-generation.yaml [hunk 1]
```diff
@@ -0,0 +1,242 @@
+name: Speakeasy SDK Generation Workflow
```
Comment (anuraagnalluri) [inline .github/workflows/sdk-docs-generation.yaml:1]:
```suggestion
name: Speakeasy SDK Docs Generation Workflow
```

File: .github/workflows/sdk-docs-generation.yaml [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            - csharp
+            - unity
+
+        required: true
+        type: string
+      mode:
+        description: |-
+          The mode to run the workflow in, valid options are 'direct' or 'pr', defaults to 'direct'.
+            - 'direct' will create a commit with the changes to the SDK Docs and push them directly to the branch the workflow is configure to run on (normally 'main' or 'master').
+            - 'pr' will instead create a new branch to commit the changes to the SDKs to and then create a PR from this branch.
```
Comment (anuraagnalluri) [inline .github/workflows/sdk-docs-generation.yaml:69]:
```suggestion
            - 'pr' will instead create a new branch to commit the changes to the SDK docs to and then create a PR from this branch.
```

File: .github/workflows/sdk-docs-generation.yaml [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+          branch_name: ${{ needs.generate.outputs.branch_name }}
+          previous_gen_version: ${{ needs.generate.outputs.previous_gen_version }}
+          speakeasy_server_url: ${{ inputs.speakeasy_server_url }}
+      - uses: ravsamhq/notify-slack-action@v2
+        if: always() && env.SLACK_WEBHOOK_URL != ''
+        with:
+          status: ${{ job.status }}
+          token: ${{ secrets.github_access_token }}
+          notify_when: "failure"
+          notification_title: "SDK Finalization Failed"
```
Comment (anuraagnalluri) [inline .github/workflows/sdk-docs-generation.yaml:238]:
```suggestion
          notification_title: "SDK Docs Finalization Failed"
```

File: .github/workflows/sdk-docs-generation.yaml [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+      - generate
+      - compile-docs
+    runs-on: ubuntu-latest
+    outputs:
+      commit_hash: ${{ steps.finalize.outputs.commit_hash }}
+    steps:
+      - name: Tune GitHub-hosted runner network
+        uses: smorimoto/tune-github-hosted-runner-network@v1
+      - id: Finalize
+        uses: speakeasy-api/sdk-generation-action@v14.33
```
Comment (anuraagnalluri) [inline .github/workflows/sdk-docs-generation.yaml:223]: Sure you're already aware, but just a reminder to change these back to @v14 before merging

Comment (ryan-timothy-albert) [reply .github/workflows/sdk-docs-generation.yaml:223]: Yup!

**Review by anuraagnalluri — APPROVED** [review]

**Pull request merged**